### PR TITLE
Story/rx 1750 add tests

### PR DESCRIPTION
**Pull request by giarcjc**
## What changed in the code?

Adds test coverage for OmbudUploadAdaptor plugin introduced for RX-1750, and adds a ckeditor testing lib cherry-picked from https://github.com/ckeditor/ckeditor5-core.  

## Test coverage

-

#### Additional Testing Notes

Core functionality introduced in https://github.com/growombud/ckeditor5/pull/19 is also present in this PR since that PR hasn't been merged yet. 

## Anything else before merging (i.e. why is this on hold)?

I don't think so.

### CODE REVIEW

**Comment by dunncl15** [comment]
@giarcjc I noticed two things while testing the AC locally (I haven't dug into the code yet just testing functionality).

1) Images don't seem to be rendering in webview preview. Example [here](https://recordit.co/b0GcrFLBpt).

2) When you abort the upload process by cancelling the edit before the image finishes uploading the upload success snackbar still appears (takes a second to appear). Example [here](https://recordit.co/8j5Ta5JPKr).

File: src/Ckeditor/plugins/ombud_upload_adapter/ombud_upload_adapter.js [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+  // Aborts the upload process.
+  abort() {
+    const emitUploadFinish = this.options.emitUploadFinish;
+    return emitUploadFinish(null, 'Upload aborted');
+  }
+
+  // we need to handle potential response types
+  handleResponse(response) {
+    if (response.ok) {
+      const contentType = response.headers.get('Content-Type') || '???';
```
Comment (dunncl15) [inline src/Ckeditor/plugins/ombud_upload_adapter/ombud_upload_adapter.js:95]: @giarcjc I'm curious about this fallback option `???`. Are there any potential errors that could come from this?

File: src/Ckeditor/plugins/ombud_upload_adapter/ombud_upload_adapter.js [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+      emitUploadFinish = () => {},
+    } = this.options;
+
+    emitUploadStart(file);
+
+    // Prepare the form data.
+    const data = new FormData();
+
+    // Important!!  OMBUD-specific implementation
+    // Buddha expects the name of the form to be 'file' (1st param), upload will fail otherwise
```
Comment (dunncl15) [inline src/Ckeditor/plugins/ombud_upload_adapter/ombud_upload_adapter.js:139]: +1 for helpful comments.

File: src/Ckeditor/plugins/ombud_upload_adapter/ombud_upload_adapter.test.js [hunk 3]
```diff
@@ -0,0 +1,263 @@
+import OmbudUploadAdaptor from './ombud_upload_adapter';
```
Comment (dunncl15) [inline src/Ckeditor/plugins/ombud_upload_adapter/ombud_upload_adapter.test.js:1]: @giarcjc this is awesome thanks for spearheading the testing effort!

**Comment by giarcjc** [comment]
@dunncl15 Thanks for testing the AC!  

I'm pretty certain that issue #1 (img not in webview) is only happening locally because if you look at the image url in the editor the url is 'https://development.ombud.com/asset/...` whereas in the webview the image url starts with `http://localhost:8080/asset/...`.  This seems to be the case whether using slate or ckeditor, so my suspicion is that value is hard-coded when developing locally somewhere.  Since we don't see that in feature envs or prod I expect that will be resolved once this gets in a non-dev environment.  If not we will catch it when QA-ing in F2. 

Issue #2 (image upload request is not actually aborted) is definitely a good call-out.  I need to discuss with Julie since technically this is consistent with how upload requests are handled today with slate (we don't have a mechanism for actually aborting upload requests once they are initiated) although this PR introduces an event-listener for aborted requests and uses the callback to flash the snackbar telling the user the request was aborted, when clearly it was not. I'll update this PR after discussing with her.

File: src/Ckeditor/plugins/ombud_upload_adapter/ombud_upload_adapter.js (hunk 1 above)
Comment (giarcjc) [reply src/Ckeditor/plugins/ombud_upload_adapter/ombud_upload_adapter.js:79]: Good question! There shouldn't be any errors caused by this.  If the response is 'ok' but content type cannot be gotten from the response headers then the fallback is to return response.blob() (binary data).  If that's the case then on line 134 `value` will be undefined.  But your comment did prompt me to realize that in the event the response type is bad the user wouldn't be notified, so I added the call to emitUploadFinish there to give some feedback to the user if that happens.  I also added another test to cover that.  Also FYI this handleResponse function is actually just copied from how we're handling fetch responses in the RX app currently (see src/api/client.js).

Comment (dunncl15) [reply src/Ckeditor/plugins/ombud_upload_adapter/ombud_upload_adapter.js:79]: Thanks for the clarification @giarcjc LGTM!

**Pull request merged**